gunpowder/nodes/snapshot.py

```diff
@@ -114,6 +119,11 @@ def prepare(self, request):
                 if graph_key not in deps:
                     deps[graph_key] = spec
 
+            for key in self.dataset_names.keys():
+                assert key in deps, (
```

Comment:
I'm not sure if we want to enforce this. I could imagine a case where you have a snapshot node with dataset names:
```python
{
a: "volumes/a",
b: "volumes/b",
}
```
and you want to be able to request just a or just b or both, and save whatever has been requested in a snapshot.

Reply:
What would be the reason to put it into dataset_names if you don't want to save it?

Reply:
you may only want to save it sometimes

Reply:
Wouldn't you need a second snapshot node for that, it only has a single write frequency (`every`) parameter? Or do mean that you then only have to change the request in your code and not the dataset_names as well?

Reply:
yeah I meant you would just need to change the request to modify whats being saved on each successive call to the same pipeline. Not sure if this is a common use case or if we want to support it.

Reply:
It feels a bit unintuitive to me. I would expect all items in dataset_names to be saved. The way it is now, it is an intersection of dataset_names and the union of the downstream request and additional_request.
If something is discarded upstream because it's not requested downstream even though it is in dataset_names I don't even get a warning at the moment.